File: backend/src/main/java/br/com/ifsp/classify/dtos/get/ClassSessionGetDTO.java
package br.com.ifsp.classify.dtos.get;

import java.time.LocalDateTime;

public record ClassSessionGetDTO(
        String uuid,
        SubjectTeacherGetDTO subjectTeacher,
        String classroomUuid,
        LocalDateTime startTime,
        LocalDateTime endTime,
        ReportGetDTO report,
        ClassSummaryDTO classDTO,
        StudentGetDTO student,
        String recurrenceGroupUuid,
        String status,
        String cancellationReason
) {}

File: frontend/src/shared/dtos/class-session-get-dto.ts
export interface ClassSessionGetDTO {
  uuid: string
  subjectTeacher: {
    uuidEmployee: string
    employee: string
    uuidSubject: string
    subject: string
  }
  classroom: {
    uuid: string
    name: string
  }
  startTime: string
  endTime: string
  students: {
    uuid: string
    name: string
  }[]
}
